worker/task.go

```diff
@@ -0,0 +1,201 @@
+package worker
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/taskcluster/taskcluster-client-go/queue"
+	"github.com/taskcluster/taskcluster-worker/engines"
+	"github.com/taskcluster/taskcluster-worker/plugins"
+	"github.com/taskcluster/taskcluster-worker/runtime"
+)
+
+type TaskRun struct {
+	TaskId          string                       `json:"taskId"`
+	RunId           uint                         `json:"runId"`
+	SignedDeleteUrl string                       `json:"-"`
+	TaskClaim       queue.TaskClaimResponse      `json:"-"`
+	TaskReclaim     queue.TaskReclaimResponse    `json:"-"`
+	Definition      queue.TaskDefinitionResponse `json:"-"`
+	controller      *runtime.TaskContextController
+	context         *runtime.TaskContext
+	plugin          plugins.TaskPlugin
+	log             *logrus.Entry
+	payload         interface{}
+}
+
+func (t *TaskRun) Prepare(engine engines.Engine) (<-chan engines.SandboxBuilder, <-chan error) {
+	out := make(chan engines.SandboxBuilder, 1)
+	errc := make(chan error, 1)
+	go func() {
```

Comment:
Yea let's talk about it.  One of my commits started going down the path of everything being serial, and that worked.  But then when we added error handling, and especially the case of having to abort somewhere in there made things a bit more muddled.  

I'm excited to discuss it though when you have a moment.